TargetComposition/SymBuilder/ApiProvider.cpp

```diff
@@ -371,6 +392,28 @@ Object SymbolBuilderNamespace::CreateSymbols(_In_ const Object& /*contextObject*
 
     CheckHr(spSymbolProcess->CreateSymbolsForModule(spModule.Get(), moduleKey, &spSymbolSet));
 
+    //
+    // If we have been asked to automatically import symbols, set up an appropriate "on demand" importer.
+    // It is *NOT* a failure to create the symbol builder set if we cannot set up the importer!
+    //
+    std::unique_ptr<SymbolImporter> spImporter;
+    if (autoImportSymbols)
+    {
+        //
+        // Go ask the debugger through the data model for its symbol path.
+        //
+        std::wstring symPath = (std::wstring)Object::RootNamespace().KeyValue(L"Debugger")
+                                                                    .KeyValue(L"Settings")
+                                                                    .KeyValue(L"Symbols")
+                                                                    .KeyValue(L"Sympath");
+
+        spImporter.reset(new SymbolImporter_DbgHelp(spSymbolSet.Get(), symPath.c_str()));
+        if (SUCCEEDED(spImporter->ConnectToSource()))
+        {
+            spSymbolSet->SetImporter(std::move(spImporter));
+        }

```

Comment:
Do you think it's worth checking if the symbol path changes whenever we try to query the symbol importer?

E.g. in SymbolSet::FindSymbolByName/SymbolSet::FindSymbolByOffset.

Reply:
Not when we query...  though I could see a "we couldn't find a PDB...  try again if the symbol path changes instead of being static here...

I'll put a todo comment here....